Automattic/vip-design-system
# Adds new component: NewForm/Select

## Description

Adding the `NewForm/Select` new component 🥳 . 

<img width="454" alt="Select Default" src="https://user-images.githubusercontent.com/199867/188659111-f76717d0-cb59-47e0-a823-b827e7251644.png">

<img width="410" alt="Select with Focus" src="https://user-images.githubusercontent.com/199867/188659428-08c2664a-da20-42bb-ad2c-0ee80ed233b7.png">

<img width="410" alt="Select Default with options visible" src="https://user-images.githubusercontent.com/199867/188659156-874715f1-858b-4f74-a62e-fd7c9de641b2.png">

<img width="480" alt="Select Default with Grouped options" src="https://user-images.githubusercontent.com/199867/188659187-0cd1d291-9c3b-40a8-8c39-6fad57b9f6a0.png">

<img width="746" alt="Screen Shot 2022-09-08 at 17 57 57" src="https://user-images.githubusercontent.com/3402/189224691-2f8b4a56-ac85-4bd2-9ade-4d4dfe3d6158.png">

<img width="584" alt="Screen Shot 2022-09-08 at 17 58 03" src="https://user-images.githubusercontent.com/3402/189224699-8c75484e-4a4b-4b5f-947a-b08cda63bbf9.png">


## Checklist


## Steps to Test

1. Pull down PR.
1. `npm run dev`.
1. Open Storybook - https://deploy-preview-103--vip-design-system-components.netlify.app/?path=/story/newform-select--primary
1. Check how the Select behaves similarly to the component, but with accessibility. You can use a keyboard, and navigate between options, etc.
